job_definitions/notify_suppliers_of_framework_application_events.yml

```diff
@@ -0,0 +1,79 @@
+  }
+%}
+{% set default_framework_slug = 'g-cloud-12' %}
+---
+{% for environment in environments %}
+- job:
+    name: "notify-suppliers-of-framework-application-event-{{ environment }}"
+    display-name: "Notify suppliers of new framework application event - {{ environment }}"
+    project-type: freestyle
+    disabled: true
+    description: |
+      Send email notifications to interested supplier users about a framework application event.
+      This job is triggered manually, e.g. after a framework manager admin has uploaded a new set of clarification PDFs.
+    parameters:
+      - string:
+          name: RESUME_RUN_ID
+          description: "UUID of a previously failed run to use for resending (this will skip any users who were successfully emailed)"
+      - string:
+          name: FRAMEWORK_SLUG
+          default: {{ default_framework_slug }}
+          description: "Framework to send notifications about, e.g. 'g-cloud-12'."
+      - choice:
+          name: EMAIL_TYPE
+          default: 21_new_clarification_questions_answers
+          choices:
+            - 21_new_clarification_questions_answers
+            - 22_clarification_questions_close_next_week
+            - 23_clarification_questions_close_today
+            - 24_clarification_questions_closed
+            - 26_applications_close_soon
+            - 27_applications_close_today
+            - 40_standstill_starts_today
+            - 42_standstill_has_ended
+          description: "Type of email to send e.g. 'New clarification questions and answers have been published'"
+      - bool:
+          name: DRY_RUN
+          default: false
+          description: "List notifications that would be sent without sending the emails"
+    scm:
+      - git:
+          url: git@github.com:alphagov/digitalmarketplace-scripts.git
+          credentials-id: github_com_and_enterprise
+          branches:
+            - master
+          wipe-workspace: false
+    publishers:
+      - trigger-parameterized-builds:
+          - project: notify-slack
+            condition: UNSTABLE_OR_WORSE
+            predefined-parameters: |
+              USERNAME=framework-application-event-email
+              JOB=Notify suppliers of new ${FRAMEWORK_SLUG} framework event {{ environment }}
+              ICON=:frame_with_picture:
+              STAGE={{ environment }}
+              STATUS=FAILED
+              URL=<${BUILD_URL}consoleFull|#${BUILD_NUMBER}>
+              CHANNEL=#dm-2ndline
+    builders:
+      - shell: |
+          if [ -n "$RESUME_RUN_ID" ]; then
+            FLAGS="--resume-run-id=$RESUME_RUN_ID"
+          fi
+
+          if [ "$DRY_RUN" = "true" ]; then
+            FLAGS="$FLAGS --dry-run"
+          fi
+
+          docker run -e DM_DATA_API_TOKEN_{{ environment|upper }} digitalmarketplace/scripts scripts/notify-suppliers-of-framework-application-event.py {{ environment }} $FRAMEWORK_SLUG $NOTIFY_API_TOKEN {{ notify_template_map[EMAIL_TYPE] }} $FLAGS
```

Comment:
Might want to break this up into several lines, also it's probably safer to wrap the different parameters in quote marks